tangelo/toolboxes/molecular_computation/mm_charges_solver.py

```diff
@@ -0,0 +1,161 @@
+
+        Returns:
+            List[float]: The list of charges
+            List[Tuple(float, Tuple(float, float, float))]: The atoms and geometries of the charges"""
+        pass
+
+
+def convert_files_to_pdbs(filenames: Union[str, List[str]]):
+    """Convert file or list of files to pdb files using openbabel.
+
+    Args:
+        filename (Union[str, List[str]]): The filename(s) that describe the MM region
+
+    Returns:
+        List[str]: The list of pdb equivalent filenames.
+    """
+
+    try:
+        import openbabel
+        from openbabel.openbabel import obErrorLog
+        obErrorLog.SetOutputLevel(0)
+    except ModuleNotFoundError:
+        raise ModuleNotFoundError(f"openbabel is required to use get_charges_and_coords_from_pdb_file when supplying only a pdb file."
+                                  "install with 'pip install openbabel-wheel'")
+
+    filesold = [filenames] if isinstance(filenames, str) else filenames
+
+    mol = openbabel.openbabel.OBMol()
+    conv = openbabel.openbabel.OBConversion()
+
+    files = []
+    for i, file in enumerate(filesold):
+        if file[-3:] != "pdb":
+            mol = openbabel.openbabel.OBMol()
+            conv.SetOutFormat("pdb")
+            conv.ReadFile(mol, file)
+            conv.WriteFile(mol, f"tmp{i}.pdb")
+            files += ["tmp{0}.pdb"]
+        else:
+            files += [file]
+
+    return files
+
+
+def get_default_mm_package() -> Type[MMChargesSolver]:
+    if is_package_installed("rdkit"):
+        return MMChargesSolverOpenMM()
+    elif is_package_installed("openmm"):
+        return MMChargesSolverRDKit()
+    else:
+        return None
+
+
+def get_mm_package(package: str = "default"):
+    """Return the MMCharges instance
+
+    Args:
+        package (str): The MM backend to use
+
+    Returns:
+        Type[MMChargesSolver]: The MMChargesSolver that computes the partial charges for given geometries
+    """
+    if package == "default":
+        return get_default_mm_package()
+    elif package.lower() == "openmm":
+        return MMChargesSolverOpenMM()
+    elif package.lower() == "rdkit":
+        return MMChargesSolverRDKit()
+    else:
+        raise ValueError(f"Requested MM package of {package} is not currently supported in Tangelo")
+
+
+class MMChargesSolverOpenMM(MMChargesSolver):
+    def __init__(self, force_field_params=('amber14-all.xml', 'amber14/tip3pfb.xml')):
+        self.force_field_params = force_field_params
+        pass
+
+    def get_charges(self, files) -> List[Tuple[float, Tuple[float, float, float]]]:
+        from openmm.app.pdbfile import PDBFile
+        from openmm.app.forcefield import ForceField
+        from openmm import NonbondedForce
+        from openmm.app import Modeller
+        import openbabel
+
+        pdbfiles = convert_files_to_pdbs(files)
+
+        pdbs = [PDBFile(file) for file in pdbfiles]
```

Comment:
Add a comment before that blob and tell me concisely what it builds. Seems to be about a force field and charges (you return the latter).

You can probably group L115 and 117 together, they are purely to generate handles to pdb data but don't do any computation related to your force field.